regolith/tools.py

```diff
@@ -755,3 +755,34 @@ def update_schemas(default_schema, user_schema):
             updated_schema[key] = user_schema[key]
 
     return updated_schema
+
+
+def group(db, by):
+    """
+    Group the document in the database according to the value of the doc[by] in db.
+
+    Parameters
+    ----------
+    db : generator
+        The database of documents.
+    by : basestring
+        The key to group the documents.
+
+    Returns
+    -------
+    grouped: dict
+        A dictionary mapping the feature value of group to the list of docs. All docs in the same generator have
+        the same value of doc[by].
+    """
+    grouped = {}
+    doc: dict
+    for doc in db:
+        key = doc.get(by)
+        if not key:
+            print("There is no field {} in {}".format(by, id_key(doc)))
+            continue
```

Comment:
this line not needed

Reply:
'continue' here means if the key is None then do not execute the following code and skip to the next doc. If no 'continue', it will execute the following code which leads to error. This may be misleading, so I am going to change the clause to be "if elif else" instead of using "continue".

Reply:
Yes, I think that this may be better.

Reply:
fixed.